src/server/pfs/server/driver.go

```diff
@@ -2300,25 +2300,28 @@ func (d *driver) listFile(ctx context.Context, file *pfs.File, full bool) ([]*pf
 	if err != nil {
 		return nil, err
 	}
-	paths := make(map[string]*hashtree.NodeProto, len(rootPaths))
+	seenPaths := make(map[string]struct{}, len(rootPaths))
```

Comment:
Minor point: When using a map as set, I prefer to use `map[string]bool` instead of `map[string]struct{}` since it allows you to do:

```go
if seenPaths[rootPath] {
     ...
}
```
instead of having to use `ok`.